src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -174,16 +188,81 @@ SettingsLoader::SettingsLoader(const std::string_view& userJSON, const std::stri
+        styledWriter.settings_["enableYAMLCompatibility"] = true; // suppress spaces around colons
+        styledWriter.settings_["precision"] = 6; // prevent values like 1.1000000000000001
+        jsonWriterInitialized = true;
+    }
+    return styledWriter;
+}
+
 // Generate dynamic profiles and add them to the list of "inbox" profiles
 // (meaning profiles specified by the application rather by the user).
 void SettingsLoader::GenerateProfiles()
 {
-    _executeGenerator(PowershellCoreProfileGenerator{});
-    _executeGenerator(WslDistroGenerator{});
-    _executeGenerator(AzureCloudShellGenerator{});
-    _executeGenerator(VisualStudioGenerator{});
+    auto generateProfiles = [&](const IDynamicProfileGenerator& generator) {
+        if (!_ignoredNamespaces.contains(generator.GetNamespace()))
+        {
+            _executeGenerator(generator, inboxSettings.profiles);
+        }
+    };
+
+    generateProfiles(PowershellCoreProfileGenerator{});
+    generateProfiles(WslDistroGenerator{});
+    generateProfiles(AzureCloudShellGenerator{});
+    generateProfiles(VisualStudioGenerator{});
 #if TIL_FEATURE_DYNAMICSSHPROFILES_ENABLED
-    _executeGenerator(SshHostGenerator{});
+    generateProfiles(SshHostGenerator{});
+#endif
+}
+
+// Generate ExtensionPackage objects from the profile generators.
+void SettingsLoader::GenerateExtensionPackagesFromProfileGenerators()
+{
+    auto generateExtensionPackages = [&](const IDynamicProfileGenerator& generator) {
+        std::vector<winrt::com_ptr<implementation::Profile>> profilesList;
+        _executeGenerator(generator, profilesList);
+
+        // These are needed for the FragmentSettings object
+        std::vector<Model::FragmentProfileEntry> profileEntries;
+        Json::Value profilesListJson{ Json::ValueType::arrayValue };
+
+        for (const auto& profile : profilesList)
+        {
+            const auto profileJson = profile->ToJson();
+            profilesListJson.append(profileJson);
+            profileEntries.push_back(winrt::make<FragmentProfileEntry>(profile->Guid(), hstring{ til::u8u16(Json::writeString(_getJsonStyledWriter(), profileJson)) }));
+        }
+
+        // Manually construct the JSON for the FragmentSettings object
+        Json::Value json{ Json::ValueType::objectValue };
+        json[JsonKey(ProfilesKey)] = profilesListJson;
+
+        auto generatorExtension = winrt::make_self<FragmentSettings>(hstring{ generator.GetNamespace() }, hstring{ til::u8u16(Json::writeString(_getJsonStyledWriter(), json)) }, hstring{ L"settings.json" });
+        generatorExtension->NewProfiles(winrt::single_threaded_vector<Model::FragmentProfileEntry>(std::move(profileEntries)));
+
+        auto extPkg = _registerFragment(std::move(*generatorExtension), FragmentScope::Machine);
+        extPkg->DisplayName(hstring{ generator.GetDisplayName() });
+        extPkg->Icon(hstring{ generator.GetIcon() });
+    };
+
+    // TODO CARLOS: is there a way to deduplicate this list?

```

Comment:
TODO! Decide, then commit. It's fine to have the code duplicated here, but please do add a comment explaining the duplication.